src/qt/bitcoin.cpp

```diff
@@ -364,7 +367,7 @@ void BitcoinApplication::createWindow(const NetworkStyle *networkStyle)
 
 void BitcoinApplication::createSplashScreen(const NetworkStyle *networkStyle)
 {
-    SplashScreen *splash = new SplashScreen(0, networkStyle);
+    splash = new SplashScreen(0, networkStyle);
     // We don't hold a direct pointer to the splash screen after creation, so use
     // Qt::WA_DeleteOnClose to make sure that the window will be deleted eventually.
     splash->setAttribute(Qt::WA_DeleteOnClose);
```

Comment:
You should remove this deleteonclose if you want to hold on to a pointer and/or manually delete the splash screen.

Note that this changes the lifetime of the splash to the lifetime of the entire application instead of just while it is visible.